src/Session/SessionManager.cc

```diff
@@ -139,406 +139,29 @@ void SessionManager::OnMessage(WSType* ws, std::string_view sv_message, uWS::OpC
-                    }
-                    break;
-                }
-                case CARTA::EventType::SET_VECTOR_OVERLAY_PARAMETERS: {
-                    CARTA::SetVectorOverlayParameters message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        tsk = new GeneralMessageTask<CARTA::SetVectorOverlayParameters>(session, message, head.request_id);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::STOP_FITTING: {
-                    CARTA::StopFitting message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnStopFitting(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::STOP_PV_PREVIEW: {
-                    CARTA::StopPvPreview message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnStopPvPreview(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::CLOSE_PV_PREVIEW: {
-                    CARTA::ClosePvPreview message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnClosePvPreview(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::REMOTE_FILE_REQUEST: {
-                    CARTA::RemoteFileRequest message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->OnRemoteFileRequest(message, head.request_id);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                case CARTA::EventType::CHANNEL_MAP_FLOW_CONTROL: {
-                    CARTA::ChannelMapFlowControl message;
-                    if (message.ParseFromArray(event_buf, event_length)) {
-                        session->HandleChannelMapFlowControlEvt(message);
-                        message_parsed = true;
-                    }
-                    break;
-                }
-                default: {
-                    spdlog::warn("Bad event type {}!", event_type);
-                    break;
-                }
-            }
-
-            if (!message_parsed) {
-                spdlog::warn("Bad {} message!", event_type_name);
-            }
-
-            if (tsk) {
-                ThreadManager::QueueTask(tsk);
+    if (op_code == uWS::OpCode::BINARY && sv_message.length() >= sizeof(EventHeader)) {
+        session->UpdateLastMessageTimestamp();
+
+        EventHeader head = *reinterpret_cast<const EventHeader*>(sv_message.data());
```

Comment:
Related to this:

1. `Util/Message` has a static `EventType` function which gets the type from an encoded message, but I believe it's also currently unused. Since we actually need the whole header here and not just the type, I suggest that you replace it with a static `GetEventHeader` function which returns the `EventHeader`, and add a `GetType` function to the `EventHeader` struct which returns the `CARTA::EventType`.
2. For consistency, it would be good to add an `EncodeMessage` function to `Message`, and move the encoding code from `Session::SendEvent` inside (not the compression decision; that can stay in `SendEvent` -- construct `msg_vs_compress` at the end using the final message size).